src/sage/rings/power_series_ring.py

```diff
@@ -878,6 +892,16 @@ def _element_constructor_(self, f, prec=infinity, check=True):
                                               check=check)
                 else:
                     raise TypeError("Can only convert series into ring with same variable name.")
+            else:
+                from sage.symbolic.ring import SR    
+                sym_var = SR.var(self.variable_name())
+                if f.is_polynomial(sym_var):
+                    poly_ring = self._poly_ring()
+                    return self.element_class(self, poly_ring(f), prec, check=check)
+                func = f.function(sym_var) # test for exception
+                L = LazyPowerSeriesRing(self.base_ring(), self.variable_name())
+                series_lazy = L.taylor(func)    
```

Comment:
There's another way to get the Taylor expansion

```
sage: (1/(x+1)).series(x, 10)
1 + (-1)*x + 1*x^2 + (-1)*x^3 + 1*x^4 + (-1)*x^5 + 1*x^6 + (-1)*x^7 + 1*x^8 + (-1)*x^9 + Order(x^10)
```

Are they equivalent, if not is there any significant difference (e.g. which one is faster?)